Xamarin.Forms.Platform.iOS/RendererPool.cs

```diff
@@ -148,7 +148,7 @@ void UpdateRenderers(Element newElement)
 
 				var x = (int)childRenderer.NativeView.Layer.ZPosition / 1000;
 				var element = newElementController.LogicalChildren[x] as VisualElement;
-				if (element == null)
+				if (element == null || childRenderer.Element.GetType() != element.GetType())
```

Comment:
Why do we need this? is this related with CompressLayout? isn't this check being handle here https://github.com/xamarin/Xamarin.Forms/blob/a7a90885a9f0557988d4c5373717bb48b917dd09/Xamarin.Forms.Platform.iOS/RendererPool.cs#L53

Reply:
Yes. This is best done in the place where this variable is defined. Fixed.